internal/etl/transform/transform.go

```diff
@@ -0,0 +1,122 @@
+package transform
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/PDOK/gomagpie/config"
+	"github.com/go-spatial/geom"
+	pggeom "github.com/twpayne/go-geom" // this lib has a large overlap with github.com/go-spatial/geom but we need it to integrate with postgres
+)
+
+type RawRecord struct {
+	FeatureID    int64
+	FieldValues  []any
+	Bbox         *geom.Extent
+	GeometryType string
+}
+
+type SearchIndexRecord struct {
+	FeatureID         string
+	CollectionID      string
+	CollectionVersion int
+	DisplayName       string
+	Suggest           string
+	GeometryType      string
+	Bbox              *pggeom.Polygon
+}
+
+type Transformer struct{}
+
+func (t Transformer) Transform(records []RawRecord, collection config.GeoSpatialCollection) ([]SearchIndexRecord, error) {
+	result := make([]SearchIndexRecord, 0, len(records))
+	for _, r := range records {
+		fieldValuesByName, err := slicesToMap(collection.Search.Fields, r.FieldValues)
+		if err != nil {
+			return nil, err
+		}
+		displayName, err := t.renderTemplate(collection.Search.DisplayNameTemplate, fieldValuesByName)
+		if err != nil {
+			return nil, err
+		}
+		suggestions := make([]string, 0, len(collection.Search.SuggestTemplates))
+		for _, suggestTemplate := range collection.Search.SuggestTemplates {
```

Comment:
de substitutions en alternatieve spellingen gaan geregeld worden in de zoeker, right?